crates/astria-conductor/src/block_verifier.rs

```diff
@@ -414,12 +495,136 @@ fn get_proposer(validator_set: &ValidatorSet) -> eyre::Result<Validator> {
 
 #[cfg(test)]
 mod test {
-    use std::str::FromStr;
+    use std::{
+        collections::BTreeMap,
+        str::FromStr,
+    };
 
-    use tendermint::block::Commit;
+    use astria_sequencer_validation::{
+        generate_action_tree_leaves,
+        MerkleTree,
+    };
+    use base64::engine::{
+        general_purpose::STANDARD,
+        Engine as _,
+    };
+    use tendermint::{
+        block::Commit,
+        validator,
+    };
 
     use super::*;
 
+    struct MockSequencerClient;
+
+    #[async_trait::async_trait]
+    impl SequencerClient for MockSequencerClient {
```

Comment:
OK, I understand now why the sequencer client trait was introduced above. I am in general not a big fan of mocking at the generics level because you don't up testing the actual implementation but a shadow of the implementation.

We should implement proper black box tests on conductor instead.

Reply:
yeah I agree, however implementing entire black box tests in this PR seemed a bit overkill since I only wanted to test the `validate_..` functions implemented on `BlockVerifier`, and they only use one function on `SequncerClient`. alternatively I can separate the code that uses `SequencerClient` vs just validates so that can be unit tested without the generic